### PR TITLE
RF: map to the correct max_features

### PR DESCRIPTION
**Pull request by wbo4958**
This PR maps the max_features to the correct value when it is "auto",

Basically follow below rules,

``` console
   *   - "auto": Choose automatically for task:
   *            If numTrees == 1, set to "all."
   *            If numTrees greater than 1 (forest), set to "sqrt" for classification and
   *              to "onethird" for regression.
```

### CODE REVIEW

File: src/spark_rapids_ml/tree.py [hunk 1]
```diff
@@ -198,6 +200,12 @@ def _rf_fit(
             rf_params = params[INIT_PARAMETERS_NAME]
             rf_params.pop("n_estimators")
 
+            if rf_params["max_features"] == "auto":
+                if total_trees == 1:
+                    rf_params["max_features"] = 1.0
+                else:
+                    rf_params["max_features"] = "sqrt" if is_classification else 0.33333
```
Comment (lijinf2) [inline src/spark_rapids_ml/tree.py:207]: PySpark says 'onethird' in the doc of method trainRegressor(). Is it equivalent to 0.33333 in the implementation?

Comment (eordentlich) [reply src/spark_rapids_ml/tree.py:207]: Very close.   Just put 1.0/3.0

Comment (wbo4958) [reply src/spark_rapids_ml/tree.py]: Done. Thx

**Review by leewyang — APPROVED** [review]

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**